server/pulp/plugins/distributor.py

```diff
@@ -337,3 +355,20 @@ def cancel_publish_group(self, call_request, call_report):
         :raise SystemExit: raised through a call to sys.exit()
         """
         sys.exit()
+
+    @staticmethod
+    def ensure_all_units_downloaded(repo_group):
+        """
+        Checks the database to make sure all units in the repos have been downloaded. If not, raises
+        an exception.
+
+        :param repo_group:  RepositoryGroup instance that should be checked
+        :type  repo_group:  pulp.plugins.model.RepositoryGroup
+
+        :raises exceptions.PulpCodedException:  if any unit in the group has un-downloaded files
+        """
+        bad_repo_ids = filter(lambda x: not repo_controller.has_all_units_downloaded(x),
+                              repo_group.repo_ids or [])
```

Comment:
What about leaving off the `or []`?

`RepositoryGroup` has that attribute as a required parameter to its `__init__` so I expect it to always be [] or do you know of a place where it is `None` or some other non-list value?